### DISCUSSION TITLE
Differences from ASP.NET Core minimal APIs with Kestrel?

### DISCUSSION THREAD

**Discussion by Atulin** [dotnet/WatsonWebserver#139] [General]
I stumbled upon this project by accident, and it seems really curious to me. I find myself asking the titular question, though.

Is Watson simply another implementation for the sake of having another one? Or is there something more to it, something that a usual, built-in minimal API project does not support? If so, I would expect *some* sort of a comparison *somewhere*, as is usually the case with projects like this. But I don't see Watson even *acknowledge* the existence of ASP.

**Comment by jchristn** [comment]
Hi @Atulin there was never an intent to create a comparison between the two.  I built the original version a long time ago before Kestrel was a thing with the main goal being to have a simple RESTful server that didn't rely on IIS for lightweight console and service projects.  A few years later, I had a need for the exact same use case, but without the dependency on `http.sys` which led to the development of `HttpServerLite` which later became `Watson.Lite`.  

The intent was never to compete with the mainstream paradigms provided my Microsoft.  Hope this helps.

**Reply by Atulin** [reply]
Understandable, thanks for the reply!
